### PR TITLE
Send CF Events to app and rack logs

### LINKED ISSUE #1984: Add CF failure events into an app's logs
Getting all Cloudformation events that indicate a failure during a stack update (e.g. a deploy) will give a user's better visibility into whats going on. These events will be available via `convox logs`.

### PR DESCRIPTION
**Pull request by MiguelMoll**
For better visibility, Rack will send any cloudformation event to the respected stack's LogGroup.

So whenever a rack update (or another change) occurs, you can see the events via `convox rack logs`
```
convox rack logs --filter=CloudformationEvent
2017-02-28T13:57:08Z CloudformationEvent resource=devrack status=UPDATE_IN_PROGRESS reason="UserInitiated"
2017-02-28T13:57:27Z CloudformationEvent resource=devrack status=UPDATE_COMPLETE_CLEANUP_IN_PROGRESS reason=""
2017-02-28T13:57:29Z CloudformationEvent resource=devrack status=UPDATE_COMPLETE reason=""
```

Same with an app deployment (or another change) `convox logs --app $APPNAME`
```
convox logs --filter=CloudformationEvent
2017-02-28T04:15:40Z CloudformationEvent resource=BalancerWebSecurityGroup status=CREATE_IN_PROGRESS reason="Resource creation Initiated"
2017-02-28T04:15:41Z CloudformationEvent resource=BalancerWebSecurityGroup status=CREATE_COMPLETE reason=""
2017-02-28T04:15:46Z CloudformationEvent resource=BalancerWeb status=CREATE_IN_PROGRESS reason=""
2017-02-28T04:15:47Z CloudformationEvent resource=BalancerWeb status=CREATE_IN_PROGRESS reason="Resource creation Initiated"
2017-02-28T04:15:48Z CloudformationEvent resource=BalancerWeb status=CREATE_COMPLETEreason=""
```

- [x] Correct the bucket where the lambda zip is located

Fixes #1984

### CODE REVIEW

**Comment by MiguelMoll** [comment]
To take advantage of this once the update is applied to a rack:

- For app CloudFormation events: an app would need to be re-deployed or change an app parameter
- For rack CloudFormation events: a rack parameter would have to be updated or an instance keyroll.

**Comment by MiguelMoll** [comment]
[Cloudformation events in Rack and App Logs](https://trello.com/c/EB80X3Io/2036-cloudformation-events-in-rack-and-app-logs)

**Comment by nzoschke** [comment]
## QA Checklist

- [ ] Update rack to cf-event-logs branch release
- [ ] Do rack parameter update
- [ ] Verify CF events in rack logs
- [ ] Do app deploy
- [ ] Verify CF events in app logs

File: cmd/convox/install.go [hunk 1]
```diff
@@ -436,6 +437,32 @@ func cmdInstall(c *cli.Context) error {
...
+	})
+	if err != nil {
+		return stdcli.Error(err)
+	}
+
+	if len(stack.Stacks) == 0 {
+		return stdcli.Error(fmt.Errorf("stack %s not found", *res.StackId))
+	}
+
+	for _, output := range stack.Stacks[0].Outputs {
```
Comment (nzoschke) [inline cmd/convox/install.go]: it seems like we don't need this in the installer

File: provider/aws/lambda/cfevents/bin/upload [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
```
Comment (nzoschke) [inline provider/aws/lambda/cfevents/bin/upload]: maybe a comment here?

File: provider/aws/lambda/cfevents/main.go [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+}
+
+func handle(r Record) error {
+	m, err := parseMessage(r.Sns.Message)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("m = %+v\n", m)
+
+	resp, err := cf.DescribeStacks(&cloudformation.DescribeStacksInput{
```
Comment (nzoschke) [inline provider/aws/lambda/cfevents/main.go]: we've had some problems in the past with these types of calls triggering rate limiting.

File: provider/aws/lambda/cfevents/main.go [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+	}
+
+	if logGroup == "" {
+		return fmt.Errorf("log group for %s not found", *stack.StackName)
+	}
+	fmt.Printf("logGroup = %+v\n", logGroup)
+
+	_, err = cwl.CreateLogStream(&cloudwatchlogs.CreateLogStreamInput{
+		LogGroupName:  aws.String(logGroup),
+		LogStreamName: aws.String(r.Sns.MessageId),
```
Comment (nzoschke) [inline provider/aws/lambda/cfevents/main.go]: I think creating a unique log stream for every event is ok.

**Pull request closed without merging**